src/reducers/mapReducer.js

```diff
@@ -0,0 +1,21 @@
+import * as types from '../actions/actionTypes';
+
+const initialState = {
+  markers: []
+};
+
+export default function mapReducer(state = initialState, action = {}) {
+  switch (action.type) {
+    case types.ADD_MARKER: {
+      let marker = action.value;
+      let newMarkersArray = state.markers.slice();
+      newMarkersArray.unshift(marker);
+      return {
+        ...state,
```

Comment:
It's not necessary to spread the state because you are defining all the props of the state(`markers`)